internal/impl/pure/scanner_csv.go

```diff
@@ -86,15 +109,18 @@ func (c *csvScannerCreator) Create(rdr io.ReadCloser, aFn service.AckFunc, detai
 	if c.customDelim != "" {
 		cRdr.Comma = []rune(c.customDelim)[0]
 	}
+	cRdr.FieldsPerRecord = c.expectedNumberOfFields
 
 	var headers []string
 	if c.parseHeaderRow {
 		tmpHeaders, err := cRdr.Read()
 		if err != nil {
 			return nil, err
 		}
-		headers = make([]string, len(tmpHeaders))
-		_ = copy(headers, tmpHeaders)
+		if len(c.expectedHeaders) > 0 && slices.Compare(c.expectedHeaders, tmpHeaders) != 0 {
+			return nil, errors.New("expected_headers don't match file contents")
```

Comment:
Is this supposed to be message or file contents?

```suggestion
			return nil, errors.New("expected_headers don't match message contents")
```

Reply:
Yes I think it should be file because the expected_headers will be checked before a message is created - so it really is checking against the file contents.